### PR TITLE
BCH: Country's Decision (DEC) updates

### PR DESCRIPTION
**Pull request by tahmeedshah**
https://jira.cbd.int/projects/DEVDESK/queues/custom/77/DEVDESK-540
for reference

### CODE REVIEW

**Review by sbilodeau — APPROVED** [review]

File: app/views/forms/edit/bch/directives/edit-biosafety-decision.directive.html [hunk 1]
```diff
@@ -24,21 +25,34 @@
...
+									</div>
 								</div>
 									<div class="row">
 										<div class="col-sm-12">
 											<div class="km-control-group" name="government" required caption="Country submitting the decision or communication">
+												<div class="help-block">
+													<u><i>Important note for European Union (EU) Members only</i></u>: if the decision applies to <i>all</i> EU Member States, 
+													the record must be published by the EU’s BCH focal point. The EU BCH focal point should select "European Union" 
+													in response to this question and the published record will appear in the country profiles of all EU Member States.
+												</div>
```
Comment (blaisefonseca) [inline app/views/forms/edit/bch/directives/edit-biosafety-decision.directive.html:43]: can we display this EU only message to EU member states only
image

**Review by blaisefonseca — CHANGES_REQUESTED** [review]

File: app/views/forms/edit/bch/directives/edit-biosafety-decision.directive.js [hunk 2]
```diff
@@ -82,8 +82,24 @@ function (app, _, template) {
                     }
                 }
                 
+                $scope.getCountryTerm = function (code) {
```
Comment (blaisefonseca) [inline app/views/forms/edit/bch/directives/edit-biosafety-decision.directive.js]: ` $scope.getCountryTerm ` no need to be on $scope, can be local function

File: app/views/forms/edit/bch/directives/edit-biosafety-decision.directive.js [hunk 3]
```diff
@@ -82,8 +82,24 @@ function (app, _, template) {
...
+                  $scope.isEuMember = false;
+                  thesaurusService.getTerms(solr.escape('eu'),{relations:true})
```
Comment (blaisefonseca) [inline app/views/forms/edit/bch/directives/edit-biosafety-decision.directive.js]: can you check if the current government == 'eu' than just plainly set isEUMember=true and not run the remaining code

File: app/views/forms/edit/bch/directives/edit-biosafety-decision.directive.js [hunk 4]
```diff
@@ -82,8 +82,24 @@ function (app, _, template) {
...
+                  .then(function(o) {
+                    var isExist = _.findIndex(o.narrowerTerms, function(item) {
+                       return item == code;
+                    });
+                  if(isExist>-1){
+                    $scope.isEuMember = true;
+                  } else {
+                    $scope.isEuMember = false;
+                  }
```
Comment (blaisefonseca) [inline app/views/forms/edit/bch/directives/edit-biosafety-decision.directive.js]: u can just use `$scope.isEUMember = _.has(o.narrowerTerms, {item:code})` // https://lodash.com/docs/3.10.1#some

File: app/views/forms/edit/bch/directives/edit-biosafety-decision.directive.js (hunk 2 above)
Comment (tahmeedshah) [reply app/views/forms/edit/bch/directives/edit-biosafety-decision.directive.js]: removed getCountryTerm function and simplify the onCountryChange function

File: app/views/forms/edit/bch/directives/edit-biosafety-decision.directive.js (hunk 3 above)
Comment (tahmeedshah) [reply app/views/forms/edit/bch/directives/edit-biosafety-decision.directive.js]: updated at:
https://github.com/scbd/absch.cbd.int/pull/451/commits/c39ca94a090542139baef64321d60c0d2aee8b42

File: app/views/forms/edit/bch/directives/edit-biosafety-decision.directive.js (hunk 4 above)
Comment (tahmeedshah) [reply app/views/forms/edit/bch/directives/edit-biosafety-decision.directive.js]: updated at:
https://github.com/scbd/absch.cbd.int/pull/451/commits/c39ca94a090542139baef64321d60c0d2aee8b42

**Review by blaisefonseca — CHANGES_REQUESTED** [review]
@tahmeedshah  FYI i am doing this changes

**Review by randyhoulahan — APPROVED** [review]

**Pull request merged**